### DISCUSSION TITLE
TypeRegistryDictionary doesn't have a method that takes Type and returns string.

### DISCUSSION THREAD

**Discussion by Jdance-Media** [SmartlyDressedGames/Unturned-3.x-Community#4312] [Ideas]
We should probably be able to get the proper string for each Type. Mainly because then plugins and modules can more easily create assets without having to make a new dictionary and replicate what you have. Although my honest opinion is that the wrapper class is silly. I originally wanted to loop through all of the registered asset types to apply some changes based on them, but I found myself unable to do so as the class did not allow me to access all of the values (I eventually switched to reflection because of this).

It also seems that TypeRegistryDictionary.getType does not work with any casing, which is inconsistent with how build types are read. 

Note: If you'd rather not change the TypeRegistryDictionary class when it comes to the getType method, you could always just make it so asset loading actually checks for a type name before defaulting to the registry, so plugins and modules could just specify the actual type name instead.

**Comment by SDGNelson** [comment]
Since it sounds like you are using it for asset types: It's currently only used for backwards compatibility with "v1" assets. You can put the fully qualified name instead and it will use `DatValue.ParseType`!
